backend/src/main/java/ai/verta/modeldb/authservice/PublicRoleServiceUtils.java

```diff
@@ -125,7 +129,12 @@ public void validateEntityUserWithUserInfo(
       throws InvalidProtocolBufferException {
     if (resourceId != null && !resourceId.isEmpty()) {
       if (modelDBServiceResourceTypes.equals(ModelDBServiceResourceTypes.PROJECT)) {
-        projectDAO.getProjectByID(resourceId);
+        if (!projectDAO.projectExistsInDB(resourceId)) {
```

Comment:
replaced the costly operation of generating the entire Project object by a count(*) query